messaging/nats.go

```diff
@@ -64,6 +66,16 @@ func ConfigureNatsConnection(config *NatsConfig, log *logrus.Entry) (*nats.Conn,
 
 // ConnectToNats will do a TLS connection to the nats servers specified
 func ConnectToNats(config *NatsConfig, errHandler nats.ErrHandler) (*nats.Conn, error) {
+	var err error
+	serversString := config.ServerString()
```

Comment:
personally, I'd choose to do this after the discover block. Something like:
```
if config.DiscoveryName != "" {
  var err error
  config.Servers, err = discoverNatsURLs(config.DiscoveryName)
  if err != nil {
    return nil, errors.Wrapf(err, "Failed to lookup servers for service %s", config.DiscoveryName)
  }
}
...

return nats.Connect(config.ServerString(), options...)
```

Reply:
ah, I like that